src/LatteContext/Collector/VariableCollector/SetParametersToTemplateVariableCollector.php

```diff
@@ -27,9 +27,9 @@ public function __construct(
         $this->templateTypeResolver = $templateTypeResolver;
     }
 
-    public function isSupported(Node $node): bool
```

Comment:
Is supported was implemented more smart, it (can) check not just node types

Reply:
At this point is was used just to check node types. We can put smarter code in collect method itself.

Reply:
Problem with smarter version is that we cannot optimize it. We have to call it for every single node of entire code even when it matches just handful assignments.

Reply:
Well true, but as you said before. This wasn't big optimization anyway

Reply:
It would be bigger on larger codebases I think. I would keep this version and we can add some version of isSupported as second level

Reply:
ok let's try it